test/unit/domain/consents/{ID}.test.ts

```diff
@@ -0,0 +1,368 @@
+  credentialId: '9876',
+  credentialType: 'FIDO',
+  credentialStatus: 'PENDING',
+  credentialPayload: 'string_representing_credential_payload',
+  credentialChallenge: 'string_representing_challenge_payload',
+  revokedAt: (new Date()).toISOString()
+}
+
+const retrievedConsentWrongChallenge: Consent = {
+  id: '1234',
+  status: 'ACTIVE',
+  initiatorId: 'pispa',
+  participantId: 'sfsfdf23',
+  credentialId: '9876',
+  credentialType: 'FIDO',
+  credentialStatus: 'PENDING',
+  credentialPayload: 'string_representing_credential_payload',
+  credentialChallenge: 'wrong_string_representing_challenge_payload',
+  revokedAt: (new Date()).toISOString()
+}
+
+/* Mock the retrieved scope value. */
+const retrievedScopes = [{
+  id: 123234,
+  consentId: '1234',
+  accountId: 'as2342',
+  action: 'account.getAccess'
+},
+{
+  id: 232234,
+  consentId: '1234',
+  accountId: 'as2342',
+  action: 'account.transferMoney'
+},
+{
+  id: 234,
+  consentId: '1234',
+  accountId: 'as22',
+  action: 'account.getAccess'
+}
+]
+
+const {
+  credential: {
+    challenge: {
+      signature,
+      payload: challenge
+    },
+    payload: publicKey,
+    id: requestCredentialId
+  }
+} = request.payload as UpdateCredentialRequest
+
+/* TODO, fill out later. */
+/* Mock the converted scope value. */
+const externalScopes: Scopes.ExternalScope[] = [
+  {
+    accountId: 'as2342',
+    actions: ['account.getAccess', 'account.transferMoney']
+  },
+  {
+    accountId: 'as22',
+    actions: ['account.getAccess']
+  }
+]
+
+/* Mock the ConsentCredential Value. */
+const credentialVerified: ConsentCredential = {
+  credentialType: 'FIDO',
+  credentialId: requestCredentialId,
+  credentialStatus: CredentialStatusEnum.VERIFIED,
+  credentialPayload: publicKey,
+  credentialChallenge: challenge
+}
+
+const consentId = retrievedConsent.id
+
+/* Mock the retrieved consent value. */
+const updatedConsent: Consent = {
+  id: '1234',
+  status: 'ACTIVE',
+  initiatorId: 'pispa',
+  participantId: 'sfsfdf23',
+  credentialId: '9876',
+  credentialType: 'FIDO',
+  credentialStatus: CredentialStatusEnum.VERIFIED,
+  credentialPayload: publicKey,
+  credentialChallenge: 'string_representing_challenge_payload'
+}
+
+// Mock Outgoing Request Body
+const requestBody: SDKStandardComponents.PutConsentsRequest = {
+  requestId: consentId,
+  scopes: externalScopes,
+  initiatorId: retrievedConsent.initiatorId as string,
+  participantId: retrievedConsent.participantId as string,
+  credential: {
+    id: requestCredentialId,
+    credentialType: 'FIDO',
+    status: CredentialStatusEnum.VERIFIED,
+    challenge: {
+      payload: retrievedConsent.credentialChallenge as string,
+      signature
+    },
+    payload: publicKey
+  }
+}
+
+describe('server/domain/consents/{ID}', (): void => {
+  beforeAll((): void => {
+    mockLoggerError.mockReturnValue(null)
+    mockLoggerPush.mockReturnValue(null)
+
+    mockConsentDbRetrieve.mockResolvedValue(retrievedConsent)
```

Comment:
so i guess the logic here is to declare best case mocks here - and then for each test, just change mocking for needed functions for that test

which prevents us from having to redeclare every mock for each test - i thought this might improve readability? Though I do see your point here